Add Basic Concepts section to docs (#78)
slackapi/bolt-python: docs/_basic/opening_modals.md

---
title: Opening modals
lang: en
slug: opening-modals
order: 10
---

<div class="section-content">
<a href="https://api.slack.com/block-kit/surfaces/modals">Modals</a> are focused surfaces that allow you to collect user data and display dynamic information. You can open a modal by passing a valid <code>trigger_id</code> and a <a href="https://api.slack.com/reference/block-kit/views">view payload</a> to the built-in client's <a href="https://api.slack.com/methods/views.open"><code>views.open</code></a> method.

> Your app receives <code>trigger_id</code>s in payloads sent to your Request URL triggered user invocation like a slash command, button press, or interaction with a select menu.

Reviewer: ```suggestion
Your app receives <code>trigger_id</code>s in payloads sent to your Request URL that are triggered by user invocations, like a shortcut, button press, or interaction with a select menu.
```

Read more about modal composition in the <a href="https://api.slack.com/surfaces/modals/using#composing_views">API documentation</a>.

</div>

```python
# Listen for a slash command invocation

> @app.command("/ticket")

Reviewer: Could we use a shortcut invocation?

> def open_modal(ack, payload, client):
>     # Acknowledge the command request
>     ack();
> 
>     # Call views_open with the built-in client
>     client.views_open({
>         # Pass a valid trigger_id within 3 seconds of receiving it
>         trigger_id=payload["trigger_id"],

Reviewer 2: ```suggestion
def open_modal(ack, body, client):
    # Acknowledge the command request
    ack();

    # Call views_open with the built-in client
    client.views_open({
        # Pass a valid trigger_id within 3 seconds of receiving it
        trigger_id=body["trigger_id"],
```
